lcm-lua/CMakeLists.txt

```diff
@@ -9,7 +9,6 @@ set(lcm_lua_sources
 
 add_library(lcm-lua MODULE ${lcm_lua_sources})
 set_target_properties(lcm-lua PROPERTIES OUTPUT_NAME "lcm" PREFIX "")
-target_compile_options(lcm-lua PRIVATE -Werror -Wall -Wextra)
```

Comment:
This line should only add `-Werror -Wall -Wextra` to the `lcm-lua` target. The `PRIVATE` should change only the target's `COMPILE_OPTIONS` (and not `INTERFACE_COMPILE_OPTIONS`). I turned this on because I fixed warnings in `lcm-lua`, and I wanted to make sure other people would not reintroduce warnings.

Can you please explain why you would want to use `--compile-no-warning-as-error`? I'm having a hard time thinking of a situation where this would be beneficial.

Reply:
I thought about this some more, and if we want to remove `-Werror` here we can do so, as long as `-Werror` is guaranteed to be turned on during CI. It would accomplish the same goal. Either way, I would prefer we keep the other options `-Wall -Wextra` here. (I do not really agree with the idea of local development using different flags from CI, but I understand the reasoning laid out in https://github.com/lcm-proj/lcm/issues/457#issuecomment-1573948863.)

Reply:
That makes a lot of sense. I agree that we should only remove the flags here that are in duplicate to what CI runs with, which excludes `-Wall -Wextra`. I've amended the original commit to [2b1d725](https://github.com/lcm-proj/lcm/commit/2b1d7252fd569d6869fa78d7f8902c2be5259101).